test/integration/testHttpClient.js

```diff
@@ -0,0 +1,113 @@
+const connOptions = require('../integration/connectionOptions');
+const ConnectionConfig = require('../../lib/connection/connection_config');
+const NodeHttpClient = require('../../lib/http/node').NodeHttpClient;
+const { hangWebServerUrl } = require('../hangWebserver');
+const assert = require('assert');
+const testUtil = require('./testUtil');
+
+describe('HttpClient Tests', () => {
+  let httpClientInstance;
+
+  const connectionOptions = {
+    ...(connOptions.valid),
+    timeout: 100000
+  };
+
+  const httpConnectionOptions = new ConnectionConfig(connectionOptions, false, false, {
+    version: '1',
+    environment: process.versions,
+  });
+
+  beforeEach(() => {
+    httpClientInstance = new NodeHttpClient(httpConnectionOptions);
+  });
+
+  describe('Aborting requests', () => {
+    it('should allow aborting any request immediately', async () => {
+      const requestObject = httpClientInstance.request({
+        url: hangWebServerUrl + '/hang',
+        method: 'GET',
+        callback: (err) => {
+          // We expect an error due to aborting the request.
+          if (err) {
+            testUtil.isRequestCancelledError(err);
+            assert.equal(err.message, 'canceled');
+            assert.equal(err.name, 'CanceledError');
+            assert.equal(err.code, 'ERR_CANCELED');
+          } else {
+            throw Error('Expected an error from aborted request, but got success.');
```

Comment:
throwing error in callback will fail the test? questio nalso in other calls with callback

Reply:
That's right - now I incorporated everywhere the approach from the other test (with assignment of an error from callback to the variable from the outer scope).